### DISCUSSION TITLE
Concurrent mode support

### DISCUSSION THREAD

**Discussion by CarusoX** [pmndrs/zustand#960] [Q&A] [answered]
Hello, I wanted to ask if there is any plan to fully support concurrent mode in the library. I found out [here](https://github.com/dai-shi/will-this-react-global-state-work-in-concurrent-rendering) that Zustand is failing two test cases (points 5 and 6) regarding useTransition.

Also I don't know if you have any external resources or extra information to know in which cases an application will be affected if these 2 issues are not really possible to solve.

**Comment by dai-shi** [comment, marked as answer]
Failing two test cases in that tool is an **expected** behavior, and it's how React team suggests with useSyncExternalStore.
So, the answer is no, there's no plan/need for it.

In practice, it may not be an actual issue for most use cases. zustand is primarily used for global states and global states are something you want to update immediately and consistently.
I don't believe you run into any issues with the current behavior, and the worst case is just "slow", nothing will break. There are many workarounds including `useDefferedValue`. This is a new field and I don't think there are a lot of resources.
